### PR TITLE
Add a prefix option to e2e for use with resources

### PR DESCRIPTION
**Pull request by ghost**
This is helpful when cleaning manually up after a failed test in
shared a gce environment.

Signed-off-by: Sami Wagiaalla swagiaal@redhat.com

### CODE REVIEW

**Comment by quinton-hoole** [comment]
ok to test

**Comment by quinton-hoole** [comment]
I think that we need to apply this consistently across all e2e tests, or not at all.  In this PR it is documented as  "A prefix to be added to cloud resources created during testing" and yet it is only applied to persistent disks?  This would be too confusing to be useful.

**Comment by swagiaal** [comment]
> I think that we need to apply this consistently across all e2e tests, or not at all. In this PR it is documented as "A prefix to be added to cloud resources created during testing" and yet it is only applied to persistent disks? This would be too confusing to be useful.

Yeah the idea was that it gets applied incrementally as test writers came across resources where this would be useful, but let me go through and see where else this can/should be applied and I will add them to this PR.

**Comment by swagiaal** [comment]
@quinton-hoole so in terms of cloud resources created by tests I could only find that PD instance. Other cloud stuff like master and node vms are created using KUBE_GCE_INSTANCE_PREFIX.

Do you think this would be useful to apply to resources created within the cluster ? (namespaces, pods, etc) ?

Another option is to rename the option to --kube-gce-instance-prefix that way it is less generic, consistent with the environment variable, and less confusing.

**Comment by quinton-hoole** [comment]
LGTM

**Comment by quinton-hoole** [comment]
I had not realized that PD's are currently the only logical candidate, sorry.  LGTM.

**Comment by quinton-hoole** [comment]
And to answer your question, no, I doubt that in-cluster resources would be very useful, as test clusters are typically torn down.  GCE projects, and the things in them however, stay around until something or someone deletes them.

**Comment by krousey** [comment]
@swagiaal Can you rebase?

**Comment by swagiaal** [comment]
@krousey rebased and reran PD e2e test
@quinton-hoole thanks for reviewing!

File: hack/ginkgo-e2e.sh [hunk 1]
```diff
@@ -105,5 +105,6 @@ export PATH=$(dirname "${e2e_test}"):"${PATH}"
   --repo-root="${KUBE_VERSION_ROOT}" \
   --node-instance-group="${NODE_INSTANCE_GROUP:-}" \
   --num-nodes="${NUM_MINIONS:-}" \
+  --prefix="${KUBE_GCE_INSTANCE_PREFIX-e2e}" \
```
Comment (quinton-hoole) [inline hack/ginkgo-e2e.sh]: Missing a ':' here?  You're trying to default to "e2e" right?

**Comment by quinton-hoole** [comment]
Removed lgtm until last comment is resolved.  Thanks @krousey for spotting!:-)

File: hack/ginkgo-e2e.sh (hunk 1 above)
Comment (swagiaal) [inline hack/ginkgo-e2e.sh]: Yes, that's right. I have added the colon

**Comment by quinton-hoole** [comment]
LGTM!  Thanks!

**Pull request merged**